src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +835,402 @@ def __try_add_error(error_list, detail):
+        remain_list_capacity = max_package_list_capacity
+        package_list_keep_five, package_list_for_truncate = (package_list[:min_packages_count], package_list[min_packages_count:]) \
+            if len(package_list) > min_packages_count else (package_list, [])
+
+        # Empty list after 2xjson.dumps have 4-5 bytes, might cause to lose 1 extra package in truncation process
+        if len(package_list) > 0:
+            remain_list_capacity = max_package_list_capacity - self.__calc_package_payload_size_on_disk(package_list_keep_five)
+
+        return package_list_keep_five, package_list_for_truncate, remain_list_capacity
+
+    def __apply_truncation_process(self, assessment_packages, installation_packages, max_package_list_capacity, low_pri_index=None):
+        """ Helper function call split patch list method then apply truncation on assessment and installation """
+
+        installation_low_pri = []
+        installation_high_pri = installation_packages
+        # cut assessment list into [:5], [5:]
+        assessment_keep_five, assessment_packages_for_truncate, remain_list_capacity = self.__cut_list_and_get_list_capacity(assessment_packages, max_package_list_capacity)
+
+        # Apply high priority (Failed, Installed) and low priority (Pending, Excluded, Not_Selected) installation logic, and keep min 5 assessment packages
+        if low_pri_index:
+            installation_high_pri = installation_packages[:low_pri_index]
+            installation_low_pri = installation_packages[low_pri_index:]
+
+        installation_high_pri_list, packages_removed_high_pri, remain_list_capacity = self.__apply_truncation(installation_high_pri, remain_list_capacity)
+        remain_assessment, packages_removed_assessment, remain_list_capacity = self.__apply_truncation(assessment_packages_for_truncate, remain_list_capacity)
+        installation_low_pri_list, packages_removed_low_pri, _ = self.__apply_truncation(installation_low_pri, remain_list_capacity)
+
+        truncated_installation_list = installation_high_pri_list + installation_low_pri_list
+        packages_removed_installation = packages_removed_high_pri + packages_removed_low_pri
+        truncated_assessment_list = assessment_keep_five + remain_assessment
+
+        return truncated_assessment_list, packages_removed_assessment, truncated_installation_list, packages_removed_installation
+
+    def __get_installation_low_pri_index(self, priority_sorted_installation_packages, low_pri_index):
+        """" Get the first index of Pending, Excluded, or Not_Selected installation packages """
+        for index, package in enumerate(priority_sorted_installation_packages):
+            package_state = package['patchInstallationState']
+            if Constants.PENDING in package_state or Constants.EXCLUDED in package_state or Constants.NOT_SELECTED in package_state:
+                low_pri_index = index
+                return low_pri_index
```

Comment:
very minor: since we are returning low_pri_index at the end of the function, we can simply break out of the loop here and let line 1038 return low_pri_index

Reply:
return index instead since we are assigning to low_pri_index line 953.